### PR TITLE
#344: language switcher added

### PR DESCRIPTION
**Pull request by artdaw**
#344: language switcher added

### CODE REVIEW

File: AtlasDemo/AtlasDemo/AppSetup.swift [hunk 1]
```diff
@@ -48,12 +53,12 @@ class AppSetup {
         }
     }
 
-    private static func prepareOptions(useSandbox useSandbox: Bool) -> Options {
+    private static func prepareOptions(useSandbox useSandbox: Bool?=true, interfaceLanguage: String?="en") -> Options {
```
Comment (bonkey) [inline AtlasDemo/AtlasDemo/AppSetup.swift]: Missing spaces around equals: ` = `

File: AtlasDemo/AtlasDemo/Main.storyboard [hunk 2]
```diff
@@ -211,13 +211,25 @@
                                     <action selector="logoutButtonTapped:" destination="4dE-k4-8mF" eventType="touchUpInside" id="zTp-5V-W5o"/>
                                 </connections>
                             </button>
+                            <segmentedControl opaque="NO" contentMode="scaleToFill" contentHorizontalAlignment="left" contentVerticalAlignment="top" segmentControlStyle="plain" selectedSegmentIndex="0" translatesAutoresizingMaskIntoConstraints="NO" id="qLo-Q0-jea">
+                                <rect key="frame" x="238" y="462" width="125" height="29"/>
+                                <segments>
+                                    <segment title="English"/>
+                                    <segment title="German"/>
```
Comment (bonkey) [inline AtlasDemo/AtlasDemo/Main.storyboard]: Maybe Deutsch? ;-)

File: AtlasDemo/AtlasDemo/ProfileViewController.swift [hunk 3]
```diff
@@ -41,9 +42,32 @@ class ProfileViewController: UIViewController {
...
+        }
+    }
+
     @IBAction func logoutButtonTapped(sender: AnyObject) {
         Atlas.logoutUser()
         self.navigationController?.popViewControllerAnimated(true)
     }
+    
+    private func getLanguageSwitcherSelectedIndex() -> Int {
+        guard let locale = AppSetup.checkout?.client.config.interfaceLocale.objectForKey(NSLocaleLanguageCode) as? String else { return 0 }
```
Comment (bonkey) [inline AtlasDemo/AtlasDemo/ProfileViewController.swift]: Move it to `AppSetup` or maybe even partially to `Config`:

```
var interfaceLanguage: String? {
  return checkout?.client.config.interfaceLocale.objectForKey(NSLocaleLanguageCode) as? String
}
```

File: AtlasDemo/AtlasDemo/ProfileViewController.swift [hunk 4]
```diff
@@ -41,9 +42,32 @@ class ProfileViewController: UIViewController {
...
+        switch locale {
```
Comment (bonkey) [inline AtlasDemo/AtlasDemo/ProfileViewController.swift]: Maybe this way? `["en", "de"].indexOf(languageCode) ?? 0`

File: AtlasDemo/AtlasDemo/ProfileViewController.swift [hunk 5]
```diff
@@ -41,9 +42,32 @@ class ProfileViewController: UIViewController {
...
     }
 
+    @IBAction func languageSwitched(sender: UISegmentedControl) {
+        switch sender.selectedSegmentIndex {
+        case 0:
+            AppSetup.switchLanguage("en")
+        case 1:
+            AppSetup.switchLanguage("de")
+        default:
+            AppSetup.switchLanguage("en")
```
Comment (bonkey) [inline AtlasDemo/AtlasDemo/ProfileViewController.swift]: Maybe?

```
case 1:
  AppSetup.switchLanguage("de")
case 0:
  fallthrough
default:
  AppSetup.switchLanguage("en")
```

File: AtlasDemo/AtlasDemo/ProfileViewController.swift (hunk 4 above)
Comment (artdaw) [inline AtlasDemo/AtlasDemo/ProfileViewController.swift]: Agree

File: AtlasDemo/AtlasDemo/ProfileViewController.swift (hunk 3 above)
Comment (artdaw) [inline AtlasDemo/AtlasDemo/ProfileViewController.swift]: Fixed

File: AtlasDemo/AtlasDemo/ProfileViewController.swift (hunk 5 above)
Comment (artdaw) [inline AtlasDemo/AtlasDemo/ProfileViewController.swift]: Fixed

File: AtlasDemo/AtlasDemo/Main.storyboard (hunk 2 above)
Comment (artdaw) [inline AtlasDemo/AtlasDemo/Main.storyboard]: Fixed

File: AtlasDemo/AtlasDemo/AppSetup.swift (hunk 1 above)
Comment (artdaw) [inline AtlasDemo/AtlasDemo/AppSetup.swift]: Fixed

**Comment by bonkey** [comment]
👍

**Comment by artdaw** [comment]
👍

**Pull request merged**